src/usb/ftdi_ft4222.c

```diff
@@ -56,96 +66,87 @@ mraa_ftdi_ft4222_init()
     ftStatus = FT_CreateDeviceInfoList(&numDevs);
     if (ftStatus != FT_OK) {
         syslog(LOG_ERR, "FT_CreateDeviceInfoList failed: error code %d\n", ftStatus);
-        mraaStatus = MRAA_ERROR_NO_RESOURCES;
-        goto init_exit;
-    }
-
-    if (numDevs == 0) {
-        syslog(LOG_ERR, "No FT4222 devices connected.\n");
         goto init_exit;
     }
 
     devInfo = calloc((size_t) numDevs, sizeof(FT_DEVICE_LIST_INFO_NODE));
     if (devInfo == NULL) {
         syslog(LOG_ERR, "FT4222 allocation failure.\n");
-        mraaStatus = MRAA_ERROR_NO_RESOURCES;
         goto init_exit;
     }
 
     ftStatus = FT_GetDeviceInfoList(devInfo, &numDevs);
+    syslog(LOG_NOTICE, "FT_GetDeviceInfoList returned %d devices\n", numDevs);
     if (ftStatus != FT_OK) {
         syslog(LOG_ERR, "FT_GetDeviceInfoList failed (error code %d)\n", (int) ftStatus);
-        mraaStatus = MRAA_ERROR_NO_RESOURCES;
         goto init_exit;
     }
-
-    /*
-            FT4222_Version ft4222Version;
-            FT4222_STATUS ft4222Status = FT4222_GetVersion(ftHandle, &ft4222Version);
-            if (FT4222_OK == ft4222Status){
-                syslog(LOG_NOTICE, "FT4222_GetVersion %08X %08X\n", ft4222Version.chipVersion,
-       ft4222Version.dllVersion);
-             } else
-                syslog(LOG_ERR, "FT4222_GetVersion failed with code %d", ft4222Status);
-    */
-
-    syslog(LOG_NOTICE, "FT_GetDeviceInfoList returned %d devices\n", numDevs);
-
+    if (numDevs < 2) {
```

Comment:
CNFMODE3 will only list 1 device. I still think we should support it and skip initializing the builtin GPIOs.